frontend/src/Buildings/buildingReducer.js

```diff
@@ -19,7 +19,9 @@ export default function buildings(state = initialState, action) {
     case ADD_BUILDING_REQUEST:
       return { ...state, isLoading: true };
     case FETCH_BUILDINGS_SUCCESS:
-      return { ...state, listOfBuildings: action.payload, isLoading: false };
+      return {
+        ...state, listOfBuildings: action.payload, isLoading: false, error: undefined,
```

Comment:
```suggestion
        ...state, listOfBuildings: action.payload, isLoading: false,
```
If you are not going to modify the error, it's unnecessary to put it here.
because the error has been included in `...state`

Reply:
but what if the error is fixed automatically by itself? the error will stay in the state, and then the components will keep rendering the error popup. :^)

Reply:
You mean when some errors happen, It will retry several times and set error to undefine when it succeeds.

Reply:
@hongtaoTang I don't really understand the reason behind this either, could you please explain more details about this?